sys/kern/proc.c

```diff
@@ -29,26 +38,43 @@ static proc_list_t proc_list = TAILQ_HEAD_INITIALIZER(proc_list);
 static proc_list_t zombie_list = TAILQ_HEAD_INITIALIZER(zombie_list);
 static pgrp_list_t pgrp_list = TAILQ_HEAD_INITIALIZER(pgrp_list);
 
-/* Pid 0 is never available, because of its special treatment by some
- * syscalls e.g. kill. */
-static bitstr_t pid_used[bitstr_size(NPROC)] = {1};
+static pgrp_t *pgrp_lookup(pgid_t pgid);
+static proc_t *proc_find_raw(pid_t pgid);
```

Comment:
`s/pgid/pid`

What `raw` does stand for? Maybe `proc_find_nolock` instead?

Reply:
`raw` means the returned process is not locked and there's no `proc_is_alive()` check.

Reply:
Since it's so similar to `pgrp_lookup` please rename it to `proc_lookup`.

Reply:
I think that will lead to confusion between `proc_find` and `proc_lookup`. I would prefer renaming `pgrp_lookup` to `pgrp_find`